src/openms/include/OpenMS/TRANSFORMATIONS/RAW2PEAK/PeakPickerHiRes.h

```diff
@@ -447,20 +449,34 @@ namespace OpenMS
 
       Size progress = 0;
       startProgress(0, input.size() + input.getChromatograms().size(), "picking peaks");
-      for (Size scan_idx = 0; scan_idx != input.size(); ++scan_idx)
+
+      if (!input.getNrSpectra() == 0)
       {
-        if (!ListUtils::contains(ms_levels_, input[scan_idx].getMSLevel()))
+        // determine type of spectral data (profile or centroided)
+        SpectrumSettings::SpectrumType original_spectrum_type = input[0].getType();
```

Comment:
mhh... so you only check the first spectrum and assume its true for all others?

Orbitrap spectra commonly have raw MS1 and centroided MS2 spectra.
If the user chose to pick both (for whatever reason), this test would fail.

Checking each spectrum is acutally not expensive sind getType() just returns a member variable, right?